sunset/code/modules/busy_space/air_traffic.dm

```diff
@@ -0,0 +1,129 @@
+	var/wrong_atc_names = list("Sol Command", "Orion Control", "[GLOB.using_map.dock_name]")
+	var/mission_noun = list("flight", "mission", "route")
+	var/request_verb = list("requesting", "calling for", "asking for")
+
+	//First response is 'yes', second is 'no'
+	var/requests = list("[GLOB.using_map.station_short] transit clearance" = list("cleared to transit", "unable to approve, contact regional on 953.5"),
+						"planetary flight rules" = list("cleared planetary flight rules", "unable to approve planetary flight rules due to traffic"),
+						"special flight rules" = list("cleared special flight rules", "unable to approve special flight rules for your traffic class"),
+						"current solar weather info" = list("sending you the relevant information via tightbeam", "cannot fulfill your request at the moment"),
+						"nearby traffic info" = list("sending you current traffic info", "no known traffic for your flight plan route"),
+						"remote telemetry data" = list("sending telemetry now", "no uplink from your ship, recheck your uplink and ask again"),
+						"refueling information" = list("sending refueling information now", "no fuel for your ship class in this sector"),
+						"a current system time sync" = list("sending time sync ping to you now", "your ship isn't compatible with our time sync, set time manually"),
+						"current system starcharts" = list("transmitting current starcharts", "request on standby due to demand"),
+						"permission to engage FTL" = list("cleared to FTL, good day", "hold position, traffic crossing"),
+						"permission to transit system" = list("cleared to transit, good day", "hold position, traffic crossing"),
+						"permission to depart system" = list("cleared to leave via flight plan route, good day", "hold position, traffic crossing"),
+						"permission to enter system" = list("good day, cleared in as published", "hold position, traffic crossing"),
+						)
+
+	//Random chance things for variety
+	var/chatter_type = "normal"
+	if(force_chatter_type)
+		chatter_type = force_chatter_type
+	else
+		chatter_type = pick(2;"emerg",5;"wrong_freq","normal") //Be nice to have wrong_lang...
+
+	var/yes = prob(90) //Chance for them to say yes vs no
+
+	var/request = pick(requests)
+	var/callname = pick(alt_atc_names)
+	var/response = requests[request][yes ? 1 : 2] //1 is yes, 2 is no
+
+	var/full_request
+	var/full_response
+	var/full_closure
+
+	switch(chatter_type)
+		if("wrong_freq")
+			callname = pick(wrong_atc_names)
+			full_request = "[callname], this is [combined_name] on a [mission] [pick(mission_noun)] to [destname], [pick(request_verb)] [request]."
+			full_response = "[combined_name], this is [GLOB.using_map.station_short] TraCon, wrong frequency. Switch to [rand(700,999)].[rand(1,9)]."
+			full_closure = "[GLOB.using_map.station_short] TraCon, copy, apologies."
+		if("wrong_lang")
+			//Can't implement this until autosay has language support
```

Comment:
If it’s not supported just don’t add it. You can come back at it later when it’s supported